lib/dependencies.js

```diff
@@ -0,0 +1,89 @@
+var schema = require('../util/schema.js')
+var BinderYmlSchema = require('../schema/binder-yml.js')
+var DockerfileSchema = require('.../schema/dockerfile.js')
+
+/**
```

Comment:
very cool, should make it much easier for others to add new kinds of dependency handling